clients/unshaded/src/main/java/tachyon/client/block/TachyonBlockStore.java

```diff
@@ -90,17 +91,31 @@ public BufferedBlockInStream getInStream(long blockId) throws IOException {
       // TODO(calvin): Fix this RPC.
       BlockInfo blockInfo = masterClient.getBlockInfo(blockId);
       // TODO(calvin): Get location via a policy.
-      if (blockInfo.locations.isEmpty()) {
+      String localHostName = NetworkAddressUtils.getLocalHostName(ClientContext.getConf());
+      boolean hasLocalWorker = mContext.hasLocalWorker();
+      NetAddress workerNetAddress = null;
+      for (BlockLocation location : blockInfo.locations) {
+        workerNetAddress = location.getWorkerAddress();
+        boolean isLocalWorker = workerNetAddress.getHost().equals(localHostName);
+        if (hasLocalWorker) {
+          if (isLocalWorker) {
+            // Prefer local block if there exists a local worker.
```

Comment:
`LocalBlockInStream` can be returned here, which makes the code path more clear, but I `break` here deliberately to keep the codes below [this line](https://github.com/amplab/tachyon/pull/1979/files#diff-d0d36b18bcff5ea3b70113a2940f830eR117) unmodified, so that when we get location via a policy in the future, we can just replace the code changes in this PR.